### PR TITLE
Avoid copying PriorityConfig and SchedulerExtender structs for every node while running priority functions

### LINKED ISSUE #71659: Scheduler - Scheduling throughput drop
There is a significant drop in scheduling throughput in ci-kubernetes-e2e-gce-scale-performance test.

scheduling_throughput

It looks like change that cause this issue was introduced between 05/11/2018 (run 249) and 06/11/2018 (run 250)

/kind bug
/sig scheduling

### PR DESCRIPTION
**Pull request by bsalamat**
**What type of PR is this?**
/kind cleanup

For lack of a better "kind", I should mark this as a clean-up, but this has caused performance regression in the scheduler.

**What this PR does / why we need it**:
There has been a performance degradation in the scheduler. We suspect it is caused after #70274. This PR fixes the issue and applies the same fix to several similar existing code with the hope that it fixes the issue.

**Which issue(s) this PR fixes**:

Fixes #

fixes (hopefully)  #71659

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig scheduling

cc/ @wojtek-t @krzysied

### CODE REVIEW

**Comment by bsalamat** [comment]
I figured what caused the performance degradation after #70274 and #70892. It was not extra copying, it was the fact that our original code used to run old-style priority functions in parallel to the map-reduce style ones.
#70892 fixed a race condition, but it was running the old-style priority functions and when they finished, it started running the new style ones. The third commit of this PR, restores the original parallelism.

Benchmarks results with the latest version of this PR:

```
goos: linux
goarch: amd64
pkg: k8s.io/kubernetes/test/integration/scheduler_perf
BenchmarkScheduling/5000Nodes/1000Pods-12         	   10000	   8932769 ns/op
PASS
ok  	k8s.io/kubernetes/test/integration/scheduler_perf	112.951s
```

File: pkg/scheduler/core/generic_scheduler.go [hunk 1]
```diff
@@ -687,22 +689,22 @@ func PrioritizeNodes(
 		}
 	})
 
-	for i, priorityConfig := range priorityConfigs {
-		if priorityConfig.Reduce == nil {
+	for i := range priorityConfigs {
+		if priorityConfigs[i].Reduce == nil {
```
Comment (krzysied) [inline pkg/scheduler/core/generic_scheduler.go]: Just out of curiosity. There will never be a situation when Function != nil and Reduce != nil, right?

Comment (bsalamat) [reply pkg/scheduler/core/generic_scheduler.go]: That's right. The Map/Reduce functions are the new style for priority functions. The old-style was using "Function". No priority function is supposed to mix the two and none of our priority functions do that.

**Comment by wojtek-t** [comment]
/lgtm

great catch!

**Review by misterikkit — COMMENTED** [review]
/lgtm

**Pull request merged**